### PR TITLE
✨ Add ListCommits and IsArchived API

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Adds 2 new APIs using graphQL.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

File: clients/repo_client.go [hunk 1]
```diff
@@ -40,39 +39,11 @@ func NewRepoUnavailableError(err error) error {
 
 type RepoClient interface {
 	InitRepo(owner, repo string) error
+	IsArchived() bool
```
Comment (laurentsimon) [inline clients/repo_client.go:42]: returning an `error` not necessary?

File: e2e/active_test.go [hunk 2]
```diff
@@ -23,18 +23,22 @@ import (
 
 	"github.com/ossf/scorecard/v2/checker"
 	"github.com/ossf/scorecard/v2/checks"
+	"github.com/ossf/scorecard/v2/clients/githubrepo"
 	scut "github.com/ossf/scorecard/v2/utests"
 )
 
 var _ = Describe("E2E TEST:Active", func() {
```
Comment (laurentsimon) [inline e2e/active_test.go:30]: use the `checker.CheckActive` variable `"E2E TEST:" + checker.CheckActive`. I've started doing this too.

**Review by laurentsimon — APPROVED** [review]

File: checks/active.go [hunk 3]
```diff
@@ -50,15 +52,10 @@ func IsActive(c *checker.CheckRequest) checker.CheckResult {
 	threshold := time.Now().In(tz).AddDate(0, 0, -1*lookBackDays)
 	totalCommits := 0
 	for _, commit := range commits {
-		commitFull, _, err := c.Client.Git.GetCommit(c.Ctx, c.Owner, c.Repo, commit.GetSHA())
-		if err != nil {
```
[the remark this answers is not in the document]
Comment (azeemshaikh38) [reply checks/active.go:54]: No line is removed here because there is no `err` returned.

File: e2e/active_test.go (hunk 2 above)
Comment (azeemshaikh38) [reply e2e/active_test.go:30]: Done.

File: clients/githubrepo/client.go [hunk 4]
```diff
@@ -66,6 +66,14 @@ func (client *Client) ListMergedPRs() ([]clients.PullRequest, error) {
 	return client.graphClient.getMergedPRs()
 }
 
+func (client *Client) ListCommits() ([]clients.Commit, error) {
+	return client.graphClient.getCommits()
+}
+
+func (client *Client) IsArchived() bool {
```
[the remark this answers is not in the document]
Comment (azeemshaikh38) [reply clients/githubrepo/client.go]: Done.

File: clients/repo_client.go (hunk 1 above)
Comment (azeemshaikh38) [reply clients/repo_client.go]: Added error

File: e2e/active_test.go (hunk 2 above)
Comment (azeemshaikh38) [reply e2e/active_test.go:30]: Done

**Pull request merged**